test/test_prims.py

```diff
@@ -752,6 +752,26 @@ def fn1(x):
             )
             self.assertFalse(includes_aten_to_copy)
 
+    @onlyCUDA
+    @skipCUDAIfRocm
+    @dtypes(torch.float16, torch.float32)
+    def test_decomposition_data_dependent_control_flow(self, device, dtype):
```

Comment:
Maybe a `masked_fill`-specific name for the moment? 

Also, please add a comment describing this test